Content.Server/Destructible/DestructibleSystem.cs

```diff
@@ -151,6 +154,49 @@ public FixedPoint2 DestroyedAt(EntityUid uid, DestructibleComponent? destructibl
             }
             return damageNeeded;
         }
+
+        /// <summary>
+        /// Takes the damage on one entity and scales it proportionally to
+        /// another entity, based on each's dead or destruction thresholds.
+        /// </summary>
+        /// <param name="original">The entity whose current damage is used.</param>
+        /// <param name="target">The entity whose thresholds are used for scaling.</param>
+        /// <param name="damage">
+        /// The scaled damage. Null if either entity didn't have a well-defined
+        /// "break" point.
+        /// </param>
+        public bool GetScaledDamage(Entity<DamageableComponent?> original,
+            EntityUid target,
+            [NotNullWhen(true)] out DamageSpecifier? damage)
+        {
+            damage = null;
+
+            // If we can't damage the target entity not much we can do
+            if (!Resolve(original, ref original.Comp, logMissing: false))
+                return false;
+
+            // Get our best guesses for what "dead/destroyed" means for each
+            if (GetDeadOrDestroyedThreshold(original) is not { } originalMax
+                || GetDeadOrDestroyedThreshold(target) is not { } targetMax)
+                return false;
+
+            damage = original.Comp.Damage / originalMax * targetMax;
+            return true;
+        }
+
+        private FixedPoint2? GetDeadOrDestroyedThreshold(EntityUid uid)
+        {
+            // Prioritize mob thresholds
+            if (_mobThreshold.TryGetDeadThreshold(uid, out var deadThreshold))
+                return deadThreshold;
+
+            // If its MaxValue treat it the same as being undefined.
+            if (TryGetDestroyedAt(uid, out var destroyedAt)
+                && destroyedAt != FixedPoint2.MaxValue)
```

Comment:
I do realize (after the fact) that this could be eventified but considering it's just two systems that'd handle it, I'm not sure it's super warranted...